otcextensions/tests/functional/sdk/dns/v2/test_zone.py

```diff
@@ -47,13 +48,15 @@ def _create_zone(self, zone_name=None, router_id=None, zone_type='public'):
                 )
             except openstack.exceptions.BadRequestException:
                 self.zone = self.client.find_zone(zone_name)
+            self.client.wait_for_zone(self.zone)
             return
         try:
             self.zone = self.client.create_zone(
                 name=zone_name
             )
         except openstack.exceptions.BadRequestException:
             self.zone = self.client.find_zone(zone_name)
+        self.client.wait_for_zone(self.zone)
```

Comment:
and what if you do wait_for_zone just once? you have zone from one or another create. You can do `wait` only once afterwards using received zone

Reply:
fixed